### ISSUE TITLE
Frontend framework support

### ISSUE THREAD

**Issue by anandbhaskaran** [stisla/stisla#32]
I just came across this project and I really love Stisla. It is clean, simple and elegant. 

However, does Stisla has plans to be compatible with front-end frameworks such as Angular, React or Vue? I believe this would be a great support for the frontend app developers.

Further introduction of Stisla UX in Adobe XD or Sketch would make the Stisla even more developer-friendly!

**Comment by nauvalazhar** [comment]
Hi, thank you for using Stisla.

In the near future, officially, I haven't thought about it and I don't plan it, just want to focus on the core, components, and infrastructure of Stisla. With that, I can work faster.

But, if anyone wants to integrate with the frontend framework, such as angular, Vue, React or something else, go ahead. I will help to put links in README that go to their repositories (such as [Rails and CodeIgniter integration](https://github.com/stisla/stisla#integrates)).

If I integrate it, I won't put it in this organization, maybe in my personal repository. Again, I don't forbid that, I will try to find someone who wants to integrate the frontend framework.

Thank you

**Comment by anandbaskaran** [comment]
Thank you @nauvalazhar! I will let you know if we plan to implement it. 

I agree that focusing on Core would be best for Stisla. However, having the basic implementation of Adobe XD or Sketch design could also be nice for the users.

**Comment by nauvalazhar** [comment]
Good if you will implement it :)

Yep, I will start thinking about it for Adobe XD and Sketch :D

**Comment by lkloon123** [comment]
I will try to make a vue integration side by side as im using this template in current vue project :)

**Comment by coderabsolute** [comment]
@lkloon123 I understand you are working on the integration with Vue, am wondering what is the status? If you need some help then I would be more than happy to contribute with my little knowledge...

**Comment by nauvalazhar** [comment]
It would be nice if @lkloon123 could create a public repository for that. So, everyone can help 😄

**Comment by cyberhck** [comment]
I'll try to work on react version for this with typescript using which you'll get intellisense on your IDE even if you don't use typescript, please let me know if anyone wants to help.

**Comment by cyberhck** [comment]
is there a adobe xd / sketch file for this?

**Comment by nauvalazhar** [comment]
Will be great if you working on react version. No, this template hasn't Adobe XD/Sketch file.

**Comment by cyberhck** [comment]
Since his is admin template, I'll try to make it opinionated yet flexible, I want people to use it out of the box without having to do anything.

**Comment by nauvalazhar** [comment]
I think to refactor the whole code to be more flexible and not depend on Bootstrap and not just the Admin dashboard.

**Comment by cyberhck** [comment]
Yes, I don't want to depend on bootstrap either, but I do want the grid though, I'll try to come up with something, first I want to make it work, but with fully planning on dropping bootstrap specially jquery stuff.

I'll first go for normal simple components, colors etc, I'll also try to do in a way that adding theme is possible, I'll let you know of my progress.

**Comment by lkloon123** [comment]
@coderabsolute @nauvalazhar 
Hi guys, sry for late reply, was busy these days.
i created a repo [vue-stisla](https://github.com/lkloon123/vue-stisla)
PR are welcome 👍

**Comment by coderabsolute** [comment]
@lkloon123 Thanks. I will take a look and please let me know how can I contribute.

@cyberhck You may also consider [Bulma](https://bulma.io/).

**Comment by cyberhck** [comment]
Hi @nauvalazhar communicating via issues here and you responding with anything you can made me think it's a really good idea to move forward, I want to do the react parts, but I need your help.

I kinda need support, as in if I'm not able to understand how something's working, I'll ask you, and I'd also like some contributions, if someone comes looking for this template, maybe you can link to ongoing development.

My plan is to split the project into 2 parts:

One where we only have components, we have documentation of each components, is automatically built and pushed to gh-pages, and it's released automatically as well.

In parallel I also want to do the template usage, something like your demo, to do that, I'll simply install the components package and try to recreate what you have.

In the mean time, I'd really appreciate if this project also took a small step towards customization.

I'd really love it if you could move any color information to a separate theme file, for example if you have `dropdown.scss`, I want you to split into `dropdown.theme.scss` and `dropdown.scss`, any color information will go to `dropdown.theme.scss`, but don't import `dropdown.theme.scss` directly into `dropdown.scss`, create a `theme.scss` in root level and import all `.theme.scss` and compile two separate files which will give you `core.css` and `theme.css`.

**Comment by lkloon123** [comment]
@coderabsolute
u can start with any component in bootstrap tab on [stisla demo](https://demo.getstisla.com/) :)

**Comment by nauvalazhar** [comment]
> Hi @nauvalazhar communicating via issues here and you responding with anything you can made me think it's a really good idea to move forward, I want to do the react parts, but I need your help.
> 
> I kinda need support, as in if I'm not able to understand how something's working, I'll ask you, and I'd also like some contributions, if someone comes looking for this template, maybe you can link to ongoing development.
> 
> My plan is to split the project into 2 parts:
> 
> One where we only have components, we have documentation of each components, is automatically built and pushed to gh-pages, and it's released automatically as well.
> 
> In parallel I also want to do the template usage, something like your demo, to do that, I'll simply install the components package and try to recreate what you have.
> 
> In the mean time, I'd really appreciate if this project also took a small step towards customization.
> 
> I'd really love it if you could move any color information to a separate theme file, for example if you have `dropdown.scss`, I want you to split into `dropdown.theme.scss` and `dropdown.scss`, any color information will go to `dropdown.theme.scss`, but don't import `dropdown.theme.scss` directly into `dropdown.scss`, create a `theme.scss` in root level and import all `.theme.scss` and compile two separate files which will give you `core.css` and `theme.css`.
> 
> do you have slack space by any chance?

Let's talk about this #58.

**Comment by coderabsolute** [comment]
> @coderabsolute
> u can start with any component in bootstrap tab on [stisla demo](https://demo.getstisla.com/) :)

@lkloon123 Can you please create a project with the Vue CLI, it may be easier to start working on it...

**Comment by cyberhck** [comment]
We're working on new framework on #58 , it's complete rewrite of this project, you can submit your concerns there, and yes, writing a CLI is something I'm thinking about, kind of one cli which can scaffold any frontend framework, and do utility stuff maybe, (like adding custom big components, for example you've published a component for stisla using stisla's scss, then one should be able to do something like `stisla add user/component` which will fetch from npm)

**Comment by nauvalazhar** [comment]
It would be better if there were some people who could help in developing in our new framework. Because I have been busy with my work for a few days 😄

**Comment by jonorozcoc** [comment]
Is there any progress on Vue integration? I would contribute on that

**Comment by lkloon123** [comment]
@jonorozcoc 
u can check [vue-stisla](https://github.com/lkloon123/vue-stisla)
but not developing atm due to they are trying to remake the whole stisla into new framework
will start after they released at least for beta,
u can send pull request on vue-cli setup as someone mention earlier :)

**Comment by cyberhck** [comment]
I need more people helping on stisla framework, I've also been very busy lately, if someone can drive it a bit more, it'd be awesome.

**Comment by nauvalazhar** [comment]
I'm also a bit busy lately, I plan to make design system first with Figma or Sketch.

**Comment by samstaron** [comment]
@lkloon123 how do I import this repo into vue?

**Comment by cyberhck** [comment]
@samstaron right now there's no official guide, I think you can simply start using it, but keep in mind that this does require jQuery and need to import css,

Bootstrap team is working on v5 which will remove jQuery as dependency, and they'll provide modules where we can import certain things, once that's done, I think @nauvalazhar will update stisla to use new components, which we can simply transform to react / vue / angular etc.

**Comment by lkloon123** [comment]
@samstaron actually u can just import the js and css then write the html and it is good to go, just bear in mind that it dont have good support for SPA since SPA dont trigger jquery onload when route change

**Comment by cyberhck** [comment]
instead of refactoring to v5, I think a XD design/zeplin design system might help much more,
then we can create web components which should be supported everywhere, or we can have multiple projects, but we can style using typestyle which works for all frontend frameworks, I've something in mind, I'll summarize in slack

**Comment by nauvalazhar** [comment]
But, what framework will we use?

**Comment by nauvalazhar** [comment]
For CSS framework, we can still use Bootstrap or Tailwind, or whatever. Also, we can use this: https://lit-element.polymer-project.org

**Comment by cyberhck** [comment]
no, actually, let's come up with design system, create the XD/zeplin/invision whatever you're comfortable with,

then we can go with something which won't make users need to load css/scss but just work everywhere.

one of the example of that would be typestyle, it'll work for everything, it'll only provide the styles though, or we don't even need to do that, just provide design systems, and people can "implement" this design systems in different framework :)

I was thinking I'll just create styles by just using typestyle which can be used on vanilla JS, React/Vue/Angular, so :)

That way we have minimal bundlesize, which is a big pain point right now,

**Comment by nauvalazhar** [comment]
I wrote a new issue [here](https://github.com/stisla/stisla/issues/154). @cyberhck

**Comment by SalahAdDin** [comment]
@anandbhaskaran Did you implement it?

**Comment by anandbhaskaran** [comment]
Hello @SalahAdDin. Unfortunately not yet :disappointed:  As Vue 3 getting released in the upcoming months, I am thinking to collaborate with the Stisla team to implement this using Vue 3. But no concrete plans yet.

**Comment by SalahAdDin** [comment]
So bad, I wanted to try it with React.

**Comment by lkloon123** [comment]
@SalahAdDin u can just import the css and use it, just it wont go well with SPA

**Comment by SalahAdDin** [comment]
The problem is to replace all _jquery_ "components".

**Comment by mjamilasfihani** [comment]
Hi, I've integrated Stisla with my Bootstrap 4 Generator for CodeIgniter 4. But I use npm. Maybe someone need it 😄

**Comment by SalahAdDin** [comment]
> Hi, I've integrated Stisla with my Bootstrap 4 Generator for CodeIgniter 4. But I use jsdelivr cdn (npm). Maybe someone needs it 😄

Still waiting for the React version.
